source/analyze-code.txt

```diff
@@ -0,0 +1,224 @@
+             ToListAsync();
+
+
+   .. tab:: {+query-api+}
+      :tabid: query-api-linq
+
+      .. code-block:: json
+
+         [{ "$match" : { 
+              "Genre" : genre, 
+              "Score" : { "$gte" : minScore }, 
+              "Title" : /titleSearchTerm/s }
+           }, 
+           { "$sort" : { "Score" : 1 } }]
+
+Analyze LINQ in Visual Studio
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To analyze your LINQ expressions in Visual Studio, perform the following actions:
+
+1. Install the {+product+} as described in the :doc:`Install </install>`
+   guide.
+2. Write a LINQ expression with the {+driver-short+}.
+3. Move your mouse over the :guilabel:`...` annotation beneath the first
+   method of your LINQ expression to display an information message that contains
+   the {+query-api+} translation.
+
+Click the following tabs to see a LINQ expression with and without an information message
+displayed:
+
+.. tabs::
+
+   .. tab:: Without Information Message
+      :tabid: without-message
+
+      .. figure:: /includes/images/linq.png
+         :alt: Screenshot of builder expression in visual studio with ellipsis annotation.
+
+   .. tab:: With Information Message
+      :tabid: with-message
+
+      .. figure:: /includes/images/linq-popup.png
+         :alt: Screenshot of builder expression in visual studio with information message displayed.
+
+
+If your LINQ expression contains an unsupported operator, the {+product+} outputs a
+``NotSupportedLinqExpression`` warning.
+
+Click the following tabs to see a code snippet containing an unsupported LINQ expression
+and the corresponding warning message displayed by the {+product+}:
+
+.. tabs::
+
+   .. tab:: Code Snippet
+      :tabid: code-snippet
+
+      The following code snippet contains the unsupported ``GetHashCode`` LINQ expression:
+
+      .. code-block:: csharp
+
+         _ = moviesCollection.Where(m => m.GetHashCode() == 1234);
+
+      The following screenshot shows the annotation displayed by the {+product+}
+      underneath the preceding code snippet in Visual Studio:
+
+      .. figure:: /includes/images/linq-unsupported.png
+         :alt: Screenshot of annotation beneath unsupported LINQ expression
+
+   .. tab:: Warning
+      :tabid: warning
+
+      The following is the warning generated by the {+product+}:
+
+      .. code-block:: text
+         :copyable: false
+
+         NotSupportedLinqExpression C# {document}.GetHashCode() is not supported.
+
+      The following screenshot shows the warning displayed in Visual Studio:
+
+      .. figure:: /includes/images/linq-unsupported-popup.png
+         :alt: Screenshot of warning displayed in Visual Studio from unsupported LINQ.
+
+.. include:: /includes/command-line.rst
+
+To view more examples of unsupported LINQ expressions, see the 
+`{+product+} Github repository <{+product-source-repo+}/tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressionsBasicTestCases.cs>`__.
+
+Analyze LINQ3
```

Comment:
LINQ3 is a bit tricky:
1. The analyzer can be configured to always use LINQ3 as mentioned in this section
2. When the analyzer configured to use LINQ2 and the driver supports LINQ3, the not supported expressions are validated with both LINQ versions, which might result in a different kind of warning: For example `_ = moviesCollection.Where(m => m.Producer.Substring(0, 6) == "Steven")`
gives `"NotSupportedLinq2Expression	Supported in LINQ3 only: db.coll.Aggregate([{ "$match" : { "$expr" : { "$eq" : [{ "$substrCP" : ["$Producer", 0, 6] }, "Steven"] } } }])"`. I think we should outline this behavior with an example.

Reply:
Added an "Important" admonition to the LINQ3 section mentioning the behavior described in point 2